test/functional/isVisible.js

```diff
@@ -12,45 +12,46 @@ describe('isVisible', () => {
     const options = new firefox.Options();
     options.headless();
 
-    const driver = new webdriver.Builder()
+    const driver = new Builder()
         .forBrowser('firefox')
         .setFirefoxOptions(options)
         .build();
 
-    it('Should return false when an element is hidden', async () => {
-        const hiddenElementIds = [
-            'not-visible-1',
-            'not-visible-2',
-            'not-visible-3',
-            'not-visible-4',
-            'not-visible-5',
-            'not-visible-6',
-            'not-visible-7',
-            'not-visible-8',
-            'not-visible-9',
-            'not-visible-10',
-            'not-visible-11',
-            'not-visible-12',
-        ];
+    async function checkVisibility(id, expected) {
+        await driver.wait(until.elementLocated(By.id(id)), WAIT_MS);
+        const isElementVisible = await driver.executeScript(`
+            ${ancestors}
+            ${isDomElement}
+            ${toDomElement}
+            return ${isVisible}(document.getElementById('${id}'));
+        `);
+        assert.equal(
+            isElementVisible,
+            expected,
+            `isVisible should return false for element with id '${id}'.`
```

Comment:
`false` → `${expected}`

Reply:
I am going to guess what you are saying here, since you have chosen not to use words... heh.

This assertion is not executing inside the page like the `driver.executeScript` call before it. Therefore, I have access to the scope of the `checkVisibility` function, where `expected` is defined.

Also, the first and second arguments into `assert.equal()` are just variables, not template literals.

If I'm missing something, please let me know.

Reply:
Oh; yes the error message if the check fails. Thanks for the catch.